### ISSUE TITLE
Git: add default keybinding for Stage / Revert / Unstage selected lines

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#93564]
Git extension needs to use the context `isInDiffEditor` and add keybindings only when that context is set.

Ctrl/Cmd + K, Cltr/Cmd + S / R / U makes good sense for defaults.

fyi @joaomoreno @jvesouza

**Comment by sportshead** [comment]
@isidorn Some shortcut confilcts:
<kbd>Cmd</kbd><kbd>K</kbd><kbd>Cmd</kbd><kbd>S</kbd> is used by show keyboard shortcuts.
<kbd>Cmd</kbd><kbd>K</kbd><kbd>Cmd</kbd><kbd>U</kbd> is used by remove line comment.

**Comment by isidorn** [comment]
@sportshead thanks for pointing out. Yeah we need to find some free keybindings.
However this would only be in the diff editor view. So even if we overwrite some like those I think it is fine.

**Comment by sportshead** [comment]
@isidorn I suppose we could overwrite `Remove Line Comment`, but I don't think it'd be best to overwrite`Show Keyboard Shortcuts`, as that's a settings page and we'll probably get an issue or too about that sometime down the road.

**Comment by ChrisPapp** [comment]
@isidorn Following discussion at #93498, I would like to work on this issue, too.

I also think that we should not override "Open Keyboard Shortcuts".
Since staging, is (in my mind) connected with "git add", maybe we could use
Ctrl/Cmd + K,Ctrl/Cmd + A

While Ctrl/Cmd + K,Ctrl/Cmd + G is also free, but the 'G' key is close to the 'R' key, so it's easier to call the Revert command by mistake.

**Comment by jvesouza** [comment]
Forgive my ignorance, but would it be possible to use something like ctrl + k ctrl + shift + s or ctrl + k alt + s?
Thanks.

**Comment by ChrisPapp** [comment]
@jvesouza @isidorn

OK, so I am sending a PR with the following keybindings:
git: Stage selected ranges ctrl + k, ctrl + shift + s 
git: Unstage selected ranges ctrl + k, ctrl + u
git: Revert selected ranges ctrl + k, ctrl + r

**Comment by ChrisPapp** [comment]
Actually
git: Stage selected ranges ctrl + k, ctrl + **alt** + s
is better because ctrl + k, ctrl + shift + s is used for Saving without Formatting.

**Comment by isidorn** [comment]
Great, thanks a lot. I have commented on the PR.
As mentioned in the PR @joaomoreno is back from vacation and this is not urgent, thus assigning to him and April milestone.

**Comment by ChrisPapp** [comment]
@isidorn @joaomoreno 
Thanks for the quick review.
However I found another possible conflict with the default keybindings.
On my VS Code version which I have downloaded and installed, 
Ctrl + K Ctrl + R - the keybinding we want to assign to Revert Selected Ranges- is used by the command "Help: Keyboard Shortcuts Reference".

However this command is not available on the version of VSCode I built from the source code, that's why I did not notice it at first.
For some reason KeybindingsReferenceAction.AVAILABLE is false and the command is not registered:
https://github.com/microsoft/vscode/blob/62c70b152a1e9eb1f918cfd1e82d49418b04def8/src/vs/workbench/browser/actions/helpActions.ts#L234-L236
